gobblin-utility/src/main/java/org/apache/gobblin/util/WriterUtils.java

```diff
@@ -100,6 +111,10 @@ public static Path getWriterStagingDir(State state, int numBranches, int branchI
     return new Path(getWriterStagingDir(state, numBranches, branchId), attemptId);
   }
 
+  public static Path getHiveDatasetWriterStagingDir(State state, int numBranches, int branchId, String attemptId) {
```

Comment:
why do you need this method given there's another `getHiveDatasetWriterStagingDir`?